### ISSUE TITLE
Cost function in backpropogation section is confusing and possibly incorrect. 

### ISSUE THREAD

**Issue by RylanYancey** [bfortuner/ml-cheatsheet#144]
Wikipedia defines the cost function MSE as follows:
![image](https://github.com/bfortuner/ml-glossary/assets/76256852/200c9614-8af5-4c7d-8202-37d36b914cfa)

Yet, the ml cheatsheet uses the following formulas in the backpropogation section.
![image](https://github.com/bfortuner/ml-glossary/assets/76256852/8d0044b1-3895-4824-b5c7-949e9fd7b64c)

This is particularly confusing since the Linear Regression and Gradient Descent section defines it correctly:
![image](https://github.com/bfortuner/ml-glossary/assets/76256852/0176630d-3235-46f2-9675-32185a749e0d)

**Comment by sprakshith** [comment]
The factor of (1/2) is often included for mathematical convenience. When you take the derivative of the cost function to perform gradient descent (a common optimization algorithm), the (1/2) cancels out when computing the gradient, simplifying the expressions and computations.

Also I believe that this cost function is just for an instance and they have not yet taken the mean in this case:

![image](https://github.com/bfortuner/ml-glossary/assets/49198294/e8930690-28fd-4cf0-be25-505320f33bd0)

Equation for Mean Squared Error with (1/2) multiplied for convenience: $$\frac{1}{2n} \sum_{i=1}^{n} \left( \hat{y}^{(i)} - y^{(i)} \right)^2$$

And then after taking the differentiation: $$\frac{1}{n} \sum_{i=1}^{n} \left( \hat{y}^{(i)} - y^{(i)} \right) \cdot x^{(i)}$$

_PS: If I am wrong please correct me, thanks._
